orca-clouddriver/src/main/java/com/netflix/spinnaker/orca/clouddriver/tasks/cluster/DetermineRollbackCandidatesTask.java

```diff
@@ -103,13 +105,14 @@ public long getBackoffPeriod() {
 
   @Override
   public long getTimeout() {
-    return TimeUnit.MINUTES.toMillis(5);
+    return TimeUnit.MINUTES.toMillis(timeout == null ? 5 : timeout);
```

Comment:
I would return 0 and override `getDynamicTimeout` instead. That one accepts a `StageExecution` as the input param. 

This way, you do not need to define the `private Integer timeout` at the class level.

Reply:
addressed